Drop running on EL 8 support

ekohl opened the pull request:
#### What changes are you introducing?

Foreman 3.13 has dropped support for EL 8. Biggest changes are the
removal of DNF modularity and upgrading from EL 8 to EL 9.

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

See the 3.13 release notes.

#### Anything else to add? (Considerations, potential downsides, alternative solutions you have explored, etc.)

At this time it's probably incomplete and that's why it's a draft.

#### Checklists

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.13/Katello 4.15
* [ ] Foreman 3.12/Katello 4.14 (Satellite 6.16)
* [ ] Foreman 3.11/Katello 4.13 (orcharhino 6.11 on EL8 only)
* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8/6.9/6.10)
* [ ] Foreman 3.8/Katello 4.10
* [ ] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [ ] Foreman 3.6/Katello 4.8
* [ ] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* We do not accept PRs for Foreman older than 3.5.

maximiliankolb commented on guides/common/modules/proc_configuring-repositories-satellite.adoc:
```diff
@@ -1,12 +1,6 @@
 .Procedure
 Select the operating system and version you are installing on:
 
```
```suggestion
```

drop this too?

maximiliankolb commented on guides/common/modules/proc_configuring-the-base-operating-system-with-offline-repositories.adoc:
```diff
@@ -1,7 +1,7 @@
 = Configuring the base operating system with offline repositories
 
-Use this procedure to configure offline repositories for {RHEL} 9 or {RHEL} 8, and {ProjectName} ISO images.
+Use this procedure to configure offline repositories for {RHEL} 9, and {ProjectName} ISO images.
```
```suggestion
Use this procedure to configure offline repositories for {RHEL} 9 and {ProjectName} ISO images.
```

maximiliankolb commented on guides/common/modules/proc_preparing-a-host-for-external-databases.adoc:
```diff
@@ -58,54 +51,6 @@ endif::[]
-====
-
-include::snip_verification-enabled-repolist.adoc[]
-endif::[]
```
I think that you have to restore line 108 "endif::[]".

maximiliankolb commented on guides/common/modules/proc_resolving-package-dependency-errors.adoc:
```diff
@@ -1,4 +1,4 @@
-[id="resolving-package-dependency-errors_{context}"]
+id="resolving-package-dependency-errors_{context}"]
```
```suggestion
[id="resolving-package-dependency-errors_{context}"]
```

small unintended change.

maximiliankolb commented on guides/common/modules/ref_system-requirements.adoc:
```diff
@@ -5,7 +5,7 @@ The following requirements apply to the networked base operating system:
 * x86_64 architecture
 ifdef::satellite[]
-* The latest version of {EL} 9 or {EL} 8
+* The latest version of {EL} 9
```
```suggestion
* The latest version of {RHEL} 9
```

(out of scope)

maximiliankolb reviewed (changes requested):
I had a look at the diff which looks very good IMHO. Some minor suggestions.

ekohl commented:
> At this time it's probably incomplete and that's why it's a draft.

To expand on that, I only looked at files which had `EL} 8` or `dnf-module`. Technically what I did was `vim $(rg 'EL\} 8|dnf-module' -l)` and reviewed each file/instance. There may be cases like `{RHEL}{nbsp}8` or otherwise well hidden versions that I didn't see.

ekohl commented:
Note I explicitly skipped over the external DB support.

ekohl commented on guides/common/modules/ref_supported-operating-systems.adoc:
```diff
@@ -14,11 +14,9 @@ The following operating systems are supported by the installer, have packages, a
 | Operating System | Architecture | Notes
```
I'm going to ignore Vale here. This is a table header that I'm not touching.

ianballou commented:
I see a few references to `Enterprise Linux 8` in https://theforeman-foreman-documentation-preview-pr-3473.surge.sh/nightly/Installing_Server/index-katello.html but I think they're minor and could be skipped.

ekohl commented:
The first 2 are good examples of things to address bit the last ome looks more like client support

asteflova reviewed (commented):
I checked the diff, all looks good, I only had one small note.

Browsing the Upgrade guides, I came across these references to upgrading to EL 9 in `con_upgrade-paths.adoc`:

* https://github.com/theforeman/foreman-documentation/blob/87cfcf2b8c1e54568a2cc8f89cc55fa8ae42cb65/guides/common/modules/con_upgrade-paths.adoc?plain=1#L17
* https://github.com/theforeman/foreman-documentation/blob/87cfcf2b8c1e54568a2cc8f89cc55fa8ae42cb65/guides/common/modules/con_upgrade-paths.adoc?plain=1#L29

Should these be reworded?

Then, browsing the Installation guide, I found these links to RHEL 8 docs in procedures to be performed on the server: 

* https://github.com/theforeman/foreman-documentation/blob/87cfcf2b8c1e54568a2cc8f89cc55fa8ae42cb65/guides/common/modules/ref_system-requirements.adoc?plain=1#L123
* https://github.com/theforeman/foreman-documentation/blob/87cfcf2b8c1e54568a2cc8f89cc55fa8ae42cb65/guides/common/modules/proc_installing-fapolicyd-on-server.adoc?plain=1#L53
* https://github.com/theforeman/foreman-documentation/blob/87cfcf2b8c1e54568a2cc8f89cc55fa8ae42cb65/guides/common/modules/proc_configuring-dynamic-dns-update-with-gss-tsig-authentication.adoc?plain=1#L11

Should these be removed?

asteflova commented:
... and now I noticed that Ian already pointed some of these out. I agree that the RHEL 8 links are non-blocking. Not sure about the references in `con_upgrade-paths.adoc`.

ekohl commented on guides/common/modules/ref_supported-operating-systems.adoc:
```diff
@@ -14,11 +14,9 @@ The following operating systems are supported by the installer, have packages, a
 | Operating System | Architecture | Notes
 ifdef::foreman-el,katello,orcharhino[]
 | {PlanningDocURL}Enterprise_Linux[{EL}] 9 | x86_64 only | EPEL is not supported.
-| {PlanningDocURL}Enterprise_Linux[{EL}] 8 | x86_64 only | EPEL is not supported.
```
I'd like to keep it out of scope for this PR. Prior to c140d0890eac01643fa945d987fa11559341ae31 it only included a single entry as well. Perhaps a good motivation to quickly support EL 10 ;)

ekohl commented:
> Should these be reworded?

They should be dropped since I dropped the procedures too. I updated the main commit

> Then, browsing the Installation guide, I found these links to RHEL 8 docs in procedures to be performed on the server:

I updated various manual links in a separate commit for easier review.

asteflova commented on guides/common/modules/con_upgrade-paths.adoc:
```diff
@@ -14,7 +14,6 @@ endif::[]
 ifndef::foreman-deb[]
 . Upgrade your {ProjectServer}:
 .. Upgrade your {ProjectServer} to {ProjectVersion}.
-.. Optional: Upgrade the operating system on your {ProjectServer} to {EL}{nbsp}9.
```
The steps on this line and line 29 actually included a note, you should probably drop the notes too.

ekohl replied:
Added as a separate commit. Please squash when merging.

asteflova commented on guides/common/modules/con_upgrade-paths.adoc:
```diff
@@ -14,24 +14,13 @@ endif::[]
 . Upgrade your {SmartProxyServers}:
 .. Upgrade all {SmartProxyServers} to {ProjectVersion}.
-.. Optional: Upgrade the operating system on your {SmartProxyServers} to {EL}{nbsp}9.
 +
```
```suggestion
```
